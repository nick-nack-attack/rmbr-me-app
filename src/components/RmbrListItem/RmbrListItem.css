.Rmbr {
    display: flex;
    flex-direction: column;
    justify-content: space-evenly;
}

.rmbr_header_wrapper {
     display: flex;
     flex-direction: row;
     justify-content: space-between;
 }

.rmbr_icons {
    display: flex;
    width: 10px;
    margin-right: 103px;

  }

#date_added_bolt {
    margin-right: 5px;
}

