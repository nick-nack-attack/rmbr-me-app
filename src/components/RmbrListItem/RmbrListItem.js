import React, { Component } from "react";
import RmbrmeContext from "../../contexts/RmbrmeContext";
import RmbrApiService from "../../services/rmbr-api-service";
import EditRmbrForm from "../EditRmbrForm/EditRmbrForm";
import './RmbrListItem.css'
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import PrettyDate from "../Utils/Utils";

export default class RmbrListItem extends Component {

    constructor(props) {
        super(props);
        this.state = {
            onDeleteRmbrSuccess: () => {},
            editFormDisplayed: false,
            autofocus: false,
            error: null,
            onCancelEdit: () => {}
        }
    };

    static contextType = RmbrmeContext;

    handleDeleteRmbr = rmbr_id => {
        this.setState({error: null});
        RmbrApiService.deleteRmbr(rmbr_id)
            .then(this.context.deleteRmbr(rmbr_id))
            .catch(res => {
                this.setState({error: 'Error deleting rmbr'})
                })
    };

    onHideEditForm = e => {
        this.setState({
            editFormDisplayed: false,
            autofocus: false
        })
    };

    handleShowEditForm = selected => {
        this.setState({
            autofocus: selected,
            editFormDisplayed: true
        })
    };

    renderRmbr = (rmbr_id) => {
        const dateCreated = this.props.rmbr.date_created
        return (
            <>
            <div className="Rmbr">
                <div className="rmbr_header_wrapper">
                    <div className="rmbr_title">
                        <h4>
                            { this.props.rmbr.rmbr_title }
                            <span className='edit_title_inline_button'>
                                <button onClick={() => this.handleShowEditForm(true)}><FontAwesomeIcon icon='pen' /></button>
                            </span>
                        </h4>
                        <h5 className='rmbr_added_label'>
                            <span><FontAwesomeIcon id='date_added_bolt' icon='bolt'/></span>
                            <span>
                            { dateCreated
                                ? PrettyDate(dateCreated)
                                : 'Rmbr'
                            }
                        </span>
                        </h5>
                    </div>
                </div>
                <div className="rmbr_description">
                    <p>{ this.props.rmbr.rmbr_text }</p>
                </div>
                <div className='rmbr_add_details_button'>
                    <button onClick={() => this.handleShowEditForm(false)}>
                        <span>
                            <FontAwesomeIcon
                            className='inline_before_icon'
                                icon='plus'
                            />
                                Add Details
                        </span>
                    </button>
                    <button
                        onClick={() => this.handleDeleteRmbr(rmbr_id)}
                        className='rmbr_list_delete-button'
                    >
                        <FontAwesomeIcon icon='trash-alt' />
                    </button>
                </div>
            </div>

                </>
        )
    };

    render() {
        const rmbr_id = this.props.id
        return (
            <li key={rmbr_id} className='Rmbr_list_item'>
                { this.state.editFormDisplayed === true
                    ?   <EditRmbrForm
                            rmbr={this.props.rmbr}
                            autofocus={ this.state.autofocus }
                            onHideEditForm={e => this.onHideEditForm(e)}
                        />
                    : this.renderRmbr(rmbr_id)
                }
            </li>
        )
    }
}

//format(new Date(this.props.rmbrArray.find(rbr => rbr.person_id === this.props.person.id).date_created), 'MMM, do Y')
//format(new Date(dateCreated), 'MMM, do Y')