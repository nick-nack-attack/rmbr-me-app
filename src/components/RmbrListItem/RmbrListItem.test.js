import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import RmbrListItem from "./RmbrListItem";

describe(`RmbrListItem component`, () => {

    const stubRmbr = {
        date_created: new Date()
    }

    it('renders RmbrListItem by default', () => {
        const wrapper = shallow(<RmbrListItem rmbr={stubRmbr}/>)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

})