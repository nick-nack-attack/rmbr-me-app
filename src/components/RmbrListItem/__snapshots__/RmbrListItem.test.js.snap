// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RmbrListItem component renders RmbrListItem by default 1`] = `
<li
  className="Rmbr_list_item"
>
  <div
    className="Rmbr"
  >
    <div
      className="rmbr_header_wrapper"
    >
      <div
        className="rmbr_title"
      >
        <h4>
          <span
            className="edit_title_inline_button"
          >
            <button
              onClick={[Function]}
            >
              <FontAwesomeIcon
                border={false}
                className=""
                fixedWidth={false}
                flip={null}
                icon="pen"
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size={null}
                spin={false}
                swapOpacity={false}
                symbol={false}
                title=""
                transform={null}
              />
            </button>
          </span>
        </h4>
        <h5
          className="rmbr_added_label"
        >
          <span>
            <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={false}
              flip={null}
              icon="bolt"
              id="date_added_bolt"
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              swapOpacity={false}
              symbol={false}
              title=""
              transform={null}
            />
          </span>
          <span>
            Jun 4
          </span>
        </h5>
      </div>
    </div>
    <div
      className="rmbr_description"
    >
      <p />
    </div>
    <div
      className="rmbr_add_details_button"
    >
      <button
        onClick={[Function]}
      >
        <span>
          <FontAwesomeIcon
            border={false}
            className="inline_before_icon"
            fixedWidth={false}
            flip={null}
            icon="plus"
            inverse={false}
            listItem={false}
            mask={null}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            swapOpacity={false}
            symbol={false}
            title=""
            transform={null}
          />
          Add Details
        </span>
      </button>
      <button
        className="rmbr_list_delete-button"
        onClick={[Function]}
      >
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon="trash-alt"
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
      </button>
    </div>
  </div>
</li>
`;
